crates/voicevox_core/src/publish.rs

```diff
@@ -12,10 +13,28 @@ use std::path::{Path, PathBuf};
 use std::sync::Mutex;
 
 // const PHONEME_LENGTH_MINIMAL: f32 = 0.01;
+const HIDDEN_SIZE: usize = 192;
 
 // static SPEAKER_ID_MAP: Lazy<BTreeMap<u32, (usize, u32)>> =
 //     Lazy::new(|| include!("include_speaker_id_map.rs").into_iter().collect());
 
+fn length_regulator(length: usize, embedded_vector: Vec<f32>, durations: &[f32]) -> Vec<f32> {
+    let mut length_regulated_vector = Vec::new();
+    for i in 0..length {
+        // numpy/pythonのroundと挙動を合わせるため、round_ties_even_を用いている
+        let regulation_size = ((durations[i] * 93.75).round_ties_even_() as usize) * 2; // 24000 / 256 = 93.75
+        let start = length_regulated_vector.len();
+        let expand_size = regulation_size * HIDDEN_SIZE;
+        length_regulated_vector.resize_with(start + expand_size, Default::default);
+        for j in (0..expand_size).step_by(HIDDEN_SIZE) {
```

Comment:
```suggestion
        for j in (0..regulation_size * expand_size).step_by(HIDDEN_SIZE) {
```

```suggestion
        for j in (0..expand_size).map(|i| HIDDEN_SIZE * i) {
```

[`Iterator::step_by`](https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.step_by)は次のような動作をします。

```rust
dbg!([0, 1, 2, 3, 4].into_iter().step_by(2).collect::<Vec<_>>());
```

```console
[src/main.rs:2] [0, 1, 2, 3, 4].into_iter().step_by(2).collect::<Vec<_>>() = [
    0,
    2,
    4,
]
```

Reply:
すみません...!意図がいまいち分かりませんでした...
https://github.com/SHAREVOX/sharevox_core/blob/9b7342d8ceb2c929c21cc3a359dce94dd1249aea/core/src/core.cpp#L441-L454
現状のコードは以上のような感じで、特に動作に差異がないように思うのですが、もう少し具体的に差異を教えていただけると嬉しいです...!

Reply:
すみません、このPRの方の`expand_size`を`regulation_size`と見間違えていました。

Reply:
問題なさそうという認識で大丈夫でしょうか...!

Reply:
はい。この部分に関しては。
(このPRの残りの部分はまだ見れてないですが...)